Simplify MPI IO in parallel vtu output

pengfej opened the pull request:
We can use write _shared to move the shared file pointer automatically after writing the header instead of moving it manually.

tjhei commented:
Pengfei, please run "make indent" in your build directory and commit the changes.

bangerth commented on source/base/data_out_base.cc:
```diff
@@ -7718,19 +7718,15 @@ DataOutInterface<dim, spacedim>::write_vtu_in_parallel(
                             header_size,
                             MPI_CHAR,
                             MPI_STATUS_IGNORE);
       AssertThrowMPI(ierr);
     }
 
-  ierr = MPI_Bcast(&header_size, 1, MPI_UNSIGNED, 0, comm);
-  AssertThrowMPI(ierr);
-
-  ierr = MPI_File_seek_shared(fh, header_size, MPI_SEEK_SET);
-  AssertThrowMPI(ierr);
+  
```
I have to admit that I do not understand the calls used here. Can you add a comment that explains what `MPI_File_write_shared` does? In particular, is it a write on the root processor couples with a seek on all processes?

tjhei replied:
Pengfei, can you add a comment in front of the write_shared that says
```
// Write the header on rank 0 and automatically move the 
// shared file pointer to the location after the header
```

tjhei replied:
And yes, @bangerth , it is kind of interesting how the library will synchronize movement of the shared file pointer without the other ranks participating in a collective call. It works, though. ;-)

bangerth replied:
That was the root of my confusion -- the other processes don't even call anything...

pengfej replied:
Sure, @tjhei ,that's reasonable. 

Thanks for the explanation!

pengfej commented:
> Pengfei, please run "make indent" in your build directory and commit the changes.

I just corrected it! And I also added the comment about write_shared

bangerth commented:
Feel free to also squash the two commits into one.

pengfej commented:
@bangerth All set, thanks for reminding!

bangerth commented:
/rebuild

tjhei commented on source/base/data_out_base.cc:
```diff
@@ -7718,19 +7718,17 @@ DataOutInterface<dim, spacedim>::write_vtu_in_parallel(
-                            MPI_CHAR,
-                            MPI_STATUS_IGNORE);
+      // Write the header on rank 0 and automatically move the
+      // shared file pointer to the location after header;
```
This should have been a ":" add the end.

tjhei commented:
@zjiaqi2018 can you confirm that this works correctly?

zjiaqi2018 commented:
> @zjiaqi2018 can you confirm that this works correctly?

I think it's correct, this is the comparison:
![image](https://user-images.githubusercontent.com/36655731/155823680-61d50349-e05e-4d61-8857-d14ae8b93998.png)

tjhei commented:
``ctest -R vtu`` passes for me as well. I think this is safe to merge.
